### ISSUE TITLE
Suggestion - 'Assassins Creed' chunk borders

### ISSUE THREAD

**Issue by generrosity** [feldim2425/MoreOverlays#48]
Heya - love your mod - its become staple in the mod packs that I see and play. 

We especially use this mod nowdays with heavier and heavier packs like FoolCraft which machines and conduit mods that get twitchy crossing chunk borders and loading out of order.

Assassins Creed has a concept of 'out of bounds' which shows as a transparent wall which is apparent only as you approach it.

I think it would be very useful to know the borders of the chunks not just the corners. Just showing a faintly tinted transparent wall that increases in opacity as you approach 2m to 0m (still very transparent even when right up next to it).

We have some very inexperienced players who are crashing my game world due to conduits crossing chunks, and they understand the "red Xs on the ground" concept, and I this would be a huge boon.

I have looked at your code briefly, and think it might be "fairly easy" to arrange using the existing rendering data? In theory its just connecting the corners, and a simple distance-opacity calculation. I'm not sure how happy you would be with someone tinkering, even in an offline-only way.

**Comment by feldim2425** [comment]
Yes it would be possible. But there is already a "wall" that shows the chunk. The Chunk Borders have 3 modes. "Hidden", "Corners" and "Grid". By default the F9 key cycles between them. You refered to the Corners mode where only red lines are shown on the chunk corners and by pressing F9 again it should also show a green grid connecting the corners of your current chunk.

So I don't see why a transparent Wall would be better. But it might be broken I got some reports that MoreOverlays doesn't work at all so I think my Code for KeyBindings is broken and it might not cycle trough the modes.

**Comment by generrosity** [comment]
Thanks feldim! The changing of modes does work correctly for us. Sorry that I haven't been as clear.

The Grid mode is very visually busy. I think you can see Corners of neighboring chunks, center of this chunk, particularly if you are only wanting warning as you get to the chunk edge . This is why I was suggesting being able to see less of the render the further away you get.

The grid mode is more difficult to use when underground, using poke holes for cabling, and tighter situations. This is why I suggest a transparent grid (a la fastrender vs wireframe in graphics programs).

Thanks again for your fast reply :)

**Comment by feldim2425** [comment]
I can implement a option to disable depth. So you would be able to see the grid trough walls.
And for the mode change: I didn't find any problems, so I hope recompiling with a newer Forge version fixes it.

**Comment by generrosity** [comment]
The changing of modes does work correctly for us. That was never a problem. 

An option to hide the centerlines would be very useful to lower their confusion. Seeing grid through walls might confuse these people more :(

Otherwise, I might mull on the idea and situational application more.

**Comment by feldim2425** [comment]
Oh a read "doesn't work" instead of "does work".
If that center line bothers you, there is a config option to disable it. You can also disable it ingame via the "Mod Options" menu on the pause screen or the "Mods" menu on the start screen. Then you just search MoreOverlays in the list and click on "Config"

I already tested the idea of the "grid trough walls" but I didn't implement the full mechanic to turn it on (or off). So that has to wait until next version.
